src/app/containers/RadioSchedule/Canonical/index.test.jsx

```diff
@@ -1,55 +1,173 @@
 import React from 'react';
+import fetchMock from 'fetch-mock';
 import { render, wait } from '@testing-library/react';
 import { matchSnapshotAsync } from '@bbc/psammead-test-helpers';
 import '@testing-library/jest-dom/extend-expect';
-import RadioSchedulesWithContext from '../utilities/testHelpers';
 import arabicRadioScheduleData from '#data/arabic/bbc_arabic_radio/schedule.json';
+import processRadioSchedule from '../utilities/processRadioSchedule';
+import CanonicalRadioSchedule from '.';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+
+const endpoint = 'https://localhost/arabic/bbc_arabic_radio/schedule.json';
+
+/* eslint-disable react/prop-types */
+const RadioScheduleWithContext = ({ initialData }) => (
+  <RequestContextProvider
+    isAmp={false}
+    pageType="frontPage"
+    service="arabic"
+    pathname="/arabic"
+    timeOnServer={Date.now()}
+  >
+    <ServiceContextProvider service="arabic">
+      <CanonicalRadioSchedule initialData={initialData} endpoint={endpoint} />
+    </ServiceContextProvider>
+  </RequestContextProvider>
+);
 
 describe('Canonical RadioSchedule', () => {
-  afterEach(() => {
-    fetch.resetMocks();
+  beforeEach(() => {
+    fetchMock.restore();
   });
 
-  it('renders correctly for a service with a radio schedule and page frequency URL', async () => {
-    fetch.mockResponse(JSON.stringify(arabicRadioScheduleData));
+  describe('With initial data', () => {
+    it('renders correctly for a service', async () => {
+      fetchMock.mock(endpoint, arabicRadioScheduleData);
+      const initialData = processRadioSchedule(
+        arabicRadioScheduleData,
+        'arabic',
+        Date.now(),
+      );
+      await matchSnapshotAsync(
+        <RadioScheduleWithContext
+          service="arabic"
+          initialData={initialData}
+          radioScheduleToggle
```

Comment:
Do we need the `service` and `radioScheduleToggle` props here and in the other instances of `RadioScheduleWithContext` below? They don't seem to be needed with the component on line 16 above

Reply:
We don't! Will remove them.

Reply:
Done!